jenkins-design-language/bin/import-material-icons.js

```diff
@@ -0,0 +1,248 @@
+            }
+            return sourcePath;
+        });
+}
+
+/**
+ * Traverse the tree starting at sourcePath, and find all the .js files.
+ *
+ * Only goes 3 levels deep, will throw if it gives up due to tree depth.
+ *
+ * Kind of ugly, but better than pulling in some npm module with 45 transitive dependencies.  Please let me know if
+ * there's a nicer way to do this! - JM
+ */
+function findSourceFiles(sourceIconsRoot) {
+    let visitedDirectories = [];
+    let allSourceFiles = [];
+
+    function recurseDir(dir, depth) {
+        // Don't get in any loops
+        if (visitedDirectories.indexOf(dir) !== -1) {
+            return;
+        }
+
+        if (depth > 3) {
+            throw new Error('findSourceFiles() - directory tree too deep');
+        }
+
+        visitedDirectories.push(dir);
+
+        return fs.readdirAsync(dir)
+            .then(filenames => {
+                // Turn the list of filenames into paths
+                const paths = filenames
+                    .filter(filename => filename[0] !== '.' && filename !== 'index.js') // Ignore index, hidden, '.' and '..'
+                    .map(filename => pathUtils.resolve(dir, filename));
+
+                // For each path...
+                const statPromises = paths.map(filePath => {
+                    return fs.statAsync(filePath) // ... get stats
+                        .then(stats => ({ filePath, stats })); // ... and assosciate with path
+                });
+
+                // Turn the array of promises into a promise of arrays.
+                return Promise.all(statPromises);
+            })
+            .then(filePathsAndStats => {
+                // Pick out all the JS files in dir, while collecting the list of subdirectories
+
+                let subDirPaths = [];
+
+                for (const { filePath, stats } of filePathsAndStats) {
+                    if (stats.isDirectory()) {
+                        subDirPaths.push(filePath);
+                    }
+                    else if (stats.isFile() && pathUtils.extname(filePath).toLowerCase() === '.js') {
+                        allSourceFiles.push(filePath);
+                    }
+                }
+
+                if (subDirPaths.length === 0) {
+                    return;
+                }
+
+                // Recurse
+                return Promise.all(subDirPaths.map(path => recurseDir(path, depth + 1)));
+            });
+    }
+
+    return recurseDir(sourceIconsRoot, 0).then(() => allSourceFiles);
+}
+
+/**
+ * Load an icon source file, remove the references to recompose/pure, then write into the JDL
+ *
+ * Very simplistic transform, and won't write anything if the file's not recognised / transformed correctly.
+ */
+function processIcon({ filePath, relativePath }, destPathRoot) {
+
+    // The patterns we're looking for. Only write the new file if all are met.
+    const pureImportRegex = /^import pure from 'recompose\/pure';\s*\n/gm;
```

Comment:
I had that originally, but then I changed it to be less flexible so we get an early warning if things are changed. I think those files are mechanically generated anyway so it seems unlikely.